registration-contracts/src/main/proto/lobby/contracts/registration/events.proto

```diff
@@ -0,0 +1,180 @@
+import "spine/money/money.proto";
+
+// Order related
+
+// The event propagated when an order is placed by a registrant.
+message OrderPlaced {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The ID of the related conference.
+    lobby.contracts.common.ConferenceId conference_id = 2;
+
+    // The order items representing the quantity of seats of different types at the conference.
+    repeated lobby.contracts.registration.SeatQuantity seat = 3;
+
+    // The expected expiration time if the reservation is not explicitly confirmed later.
+    google.protobuf.Timestamp reservation_auto_expiration = 4;
+
+    // The conference locator access code.
+    lobby.contracts.registration.ConferenceAccessCode access_code = 5;
+}
+
+// The event propagated when an order is updated.
+message OrderUpdated {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.contracts.registration.SeatQuantity seat = 2;
+}
+
+// The event propagated when an order is partially reserved.
+message OrderPartiallyReserved {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The time of the reservation expiration.
+    google.protobuf.Timestamp reservation_expiration = 2;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.contracts.registration.SeatQuantity seat = 3;
+}
+
+// The event propagated when an order reservation is completed.
+message OrderReservationCompleted {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The time of the reservation expiration.
+    google.protobuf.Timestamp reservation_expiration = 2;
+
+    // The order items representing the quantity of seats of different types.
+    repeated lobby.contracts.registration.SeatQuantity seat = 3;
+}
+
+// The event propagated when an order is expired.
+message OrderExpired {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+}
+
+// The event propagated when an order is confirmed.
+message OrderConfirmed {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+}
+
+// The event propagated when a registrant is assigned to an order.
+message OrderRegistrantAssigned {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The personal info of the registrant.
+    lobby.contracts.registration.PersonalInfo personal_info = 2;
+}
+
+// The event propagated when an order total price is calculated.
+message OrderTotalsCalculated {
+    // The ID of the order.
+    lobby.contracts.common.OrderId order_id = 1;
+
+    // The total price of the order.
+    spine.money.Money total = 2;
+
+    // The lines of the order.
+    repeated lobby.contracts.registration.SeatOrderLine order_line = 3;
+
+    // If the order is free of charge.
```

Comment:
This needs to be closer to total field, as they relate to each other.

Consider having oneof payable if there cannot be case when it's for money and free at the same time.